### ISSUE TITLE
[BUG/ISSUE] Collection times for HISTORY.rc

### ISSUE THREAD

**Issue by dylanbm** [geoschem/geos-chem#800]
There appears to be a glitch in the calculation of the simulation time for HISTORY.rc when doing a full-year simulation. I have a 1-year simulation starting at 20160801 and ending at 20170801. The 'Metrics' collection of HISTORY.rc is set to output at the end (as is the default).

However the simulation stops with a "no diagnostic output will be created" error (pasted below). The simulation length (SimLengthSec) variable is being calculated as 1 year (31536000) but Container%FileWriteAlarm is being calculated as 1 year + 1 day (31622400). And so it stops with the error trap in history_mod.F90.

The error doesn't occur for shorter simulations.
```
HISTORY (INIT): Opening ./HISTORY.rc
   31536000.0000000     
   31622400.0000000     
===============================================================================
GEOS-Chem ERROR: No diagnostic output will be created for collection: 
"Metrics"!  Make sure that the length of the simulation as specified in 
"input.geos" (check the start and end times) is not shorter than the frequency 
setting in HISTORY.rc!  For example, if the frequency is 010000 (1 hour) but 
the simulation is set up to run for only 20 minutes, then this error will occur.
 -> at History_ReadCollectionData (in module History/history_mod.F90)

 -> ERROR occurred at (or near) line    102 of the HISTORY.rc file
===============================================================================

===============================================================================
GEOS-Chem ERROR: Error encountered in "History_ReadCollectionData"!
 -> at History_Init (in module History/history_mod.F90)
===============================================================================

===============================================================================
GEOS-CHEM ERROR: Error encountered in "History_Init"!
STOP at  -> at GEOS-Chem (in GeosCore/main.F90)
===============================================================================
```

**Comment by yantosca** [comment]
Thanks for reporting this issue @dylanbm.  I've pushed commit 8192c7793a18948b46369ef99da235bcf461abf7 to the `patch/13.2.1` branch, which should fix this issue.  The problem was that the FileWriteAlarm and FileCloseAlarm were being incremented by 366 days.  Even though 2016 is a leap year, the alarm interval should be 365 days, because the start date of 20160801 is after March 1st (when we are past the leap year day).  I've modified the logic of in the alarm update subroutine to account for this.

This seems to work well but I have not validated every possible edge case (as it would be difficult to do so).  For long simulations, it is recommended to generate monthly restart files (with HISTORY.rc. collection settings `frequency: 00000100 0000001` and `duration: 00000100 000000`.  This will give you extra protection in case of an error (or computer glitch), so that you won't have to run the whole year but only from the last restart file.

I will close out this issue.  It will ship with version 13.2.1 (which will be released immediately after 13.2.0 once it clears benchmarking) as there have been a few bugs reported since the 13.2.0 code was frozen.
